lib/SILGen/SILGenPoly.cpp

```diff
@@ -675,6 +760,41 @@ ManagedValue Transform::transform(ManagedValue v,
     return std::move(result).getAsSingleValue(SGF, Loc);
   }
 
+  // - T.TangentVector to Optional<T>.TangentVector
+  // Optional<T>.TangentVector is a struct wrapping Optional<T.TangentVector>
+  // So we just need to call appropriate .init on it.
+  // However, we might have T.TangentVector == T, so we need to calculate all
+  // required types first.
+  if (CanType optionalTy = outputSubstType.getNominalParent(); // `Optional<T>`
```

Comment:
`getNominalParent` assumes that `outputSubstType` is a nominal type, we probably ought to check to make sure that's the case before accessing